betatest/forms.py

```diff
@@ -0,0 +1,205 @@
+        'password_confirmation',
+        'project_description',
+        'accept_terms',
+        'subscribe_to_updates',
+    )
+
+    full_name = forms.CharField(
+        max_length=255,
+        help_text='Example: Albus Dumbledore',
+    )
+    username = forms.RegexField(
+        regex=r'^[\w.+-]+$',
+        max_length=30,
+        help_text=('This would also be the username of the administrator '
+                   'account on the Open edX instance.'),
+        error_messages={
+            'invalid': ('Usernames may contain only letters, numbers, and '
+                        './+/-/_ characters.'),
+            'unique': 'This username is already taken.',
+        },
+    )
+    email = forms.EmailField(
+        help_text=('This is also your account name, and where we will send '
+                   'important notices.'),
+    )
+    password = forms.CharField(
+        widget=forms.PasswordInput,
+        help_text=('Pick a password for your OpenCraft account. You will be '
+                   'able to use it to login and access your account.'),
+    )
+    password_confirmation = forms.CharField(
+        widget=forms.PasswordInput,
+    )
+    accept_terms = forms.BooleanField(
+        required=True,
+        label='',
+        help_text=('I understand that this is a beta test, that bugs and '
+                   'crashes are expected, and that the instance is provided '
+                   'for free for the duration of the beta-test, without any '
+                   'guarantee.'),
+        error_messages={
+            'required': 'You must accept these terms to register.',
+        },
+    )
+
+    # This field is created automatically from the model field, but the regex
+    # validator is not copied over. We need to define the field manually so
+    # that validation will work client-side. We do this by copying from the
+    # model field (DRY).
+    _subdomain_field = Meta.model._meta.get_field('subdomain')
+    _subdomain_validator = next(v for v in _subdomain_field.validators
+                                if hasattr(v, 'regex'))
+    subdomain = forms.RegexField(
+        regex=_subdomain_validator.regex,
+        max_length=_subdomain_field.max_length,
+        label=capfirst(_subdomain_field.verbose_name),
+        help_text=_subdomain_field.help_text,
+        error_messages={
+            'invalid': _subdomain_validator.message,
+        },
+    )
+
+    # Form values will be available in the angular scope under this namespace
+    scope_prefix = 'registration'
+
+    # This form itself will be available in the angular scope under this name
+    form_name = 'form'
+
+    def __init__(self, *args, **kwargs):
+        """
+        Make sure fields are displayed in the correct order.
+        """
+        super().__init__(*args, **kwargs)
+        fields = self.fields.copy()
+        self.fields = OrderedDict()
+        for field in self.field_order:
+            self.fields[field] = fields[field]
```

Comment:
Why is this necessary?  I think fields are rendered in the order in which you define them in the form class, so you could just define them in the correct order.  (I vaguely remember they keep a global counter when a Field class is instantiated, and use that global counter for ordering.)
In any case, you don't need to copy `self.fields`, since you don't even modify it, and even if you did that would be OK, since the fields shared between all instances of the class are in `self.base_fields`.